packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -31,6 +32,8 @@ export default util.createRule<[Options], MessageIds>({
         'void is only valid as return type or type of `this` parameter.',
       invalidVoidNotReturnOrThisParamOrGeneric:
         'void is only valid as a return type or generic type variable or the type of a `this` parameter.',
+      InvalidVoidForUnion:
```

Comment:
This new `InvalidVoidForUnion` isn't used anywhere...?

Reply:
Except for the error message, I didn't touch the rest  like testcode.

Reply:
Yes, that's the issue 😄. Defining a message on its own doesn't do anything. You have to modify the rule's logic to report that message too.

Reply:
@JoshuaKGoldberg 
I modified the test case and logic, but I'm not sure if this is right